components/org.wso2.carbon.identity.application.authenticator.cognito/src/main/java/org/wso2/carbon/identity/application/authenticator/cognito/CognitoOIDCAuthenticator.java

```diff
@@ -0,0 +1,294 @@
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.exception.LogoutFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.application.authenticator.oidc.OpenIDConnectAuthenticator;
+import org.wso2.carbon.identity.application.common.model.Property;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Authenticator to handle log in ang log out of the AWS Cognito.
+ */
+public class CognitoOIDCAuthenticator extends OpenIDConnectAuthenticator {
+
+    private static final long serialVersionUID = 9058607724358986002L;
+    private static Log log = LogFactory.getLog(CognitoOIDCAuthenticator.class);
+    private String tokenEndpoint;
+    private String oAuthEndpoint;
+    private String logoutEndpoint;
+    private String userInfoEndpoint;
+
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Inside CognitoOIDCAuthenticator.canHandle()");
+        }
+        if (CognitoOIDCAuthenticatorConstants.COGNITO_LOGOUT.equals(request.getParameter
+                (CognitoOIDCAuthenticatorConstants.COGNITO_STATE)) && getContextIdentifier(request) != null) {
+            return true;
+        }
+        return super.canHandle(request);
+    }
+
+    @Override
+    public String getContextIdentifier(HttpServletRequest request) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Inside CognitoOIDCAuthenticator.getContextIdentifier()");
+        }
+        Cookie cookie =
+                FrameworkUtils.getCookie(request, CognitoOIDCAuthenticatorConstants.COGNITO_LOGOUT_STATE_COOKIE);
+        if (cookie != null) {
+            String state = cookie.getValue();
+            return state;
+        }
+        return null;
+    }
+
+    @Override
+    public String getClaimDialectURI() {
+
+        String claimDialectUri = super.getClaimDialectURI();
+        AuthenticatorConfig authConfig = FileBasedConfigurationBuilder.getInstance().getAuthenticatorBean(getName());
+        if (authConfig != null) {
+            Map<String, String> parameters = authConfig.getParameterMap();
+            if (parameters != null && parameters.containsKey(CognitoOIDCAuthenticatorConstants.
+                    CLAIM_DIALECT_URI_PARAMETER)) {
+                claimDialectUri = parameters.get(CognitoOIDCAuthenticatorConstants.CLAIM_DIALECT_URI_PARAMETER);
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Found no Parameter map for connector " + getName());
```

Comment:
log.debug("Found no Parameter map for connector  : " + getName());

note ":", 
Please have a separator for any context param value for all logs